app/utils/nonsipp/SoftDelete.scala

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils.nonsipp
+
+import pages.nonsipp.memberdetails._
+import pages.nonsipp.membertransferout.TransfersOutSectionCompleted.TransfersOutSectionCompletedUserAnswersOps
+import cats.syntax.apply._
+import config.Refined.{Max300, Max5, Max50}
+import controllers.PSRController
+import models.SchemeId.Srn
+import pages.nonsipp.receivetransfer._
+import models.requests.psr._
+import pages.nonsipp.memberpensionpayments.TotalAmountPensionPaymentsPage
+import pages.nonsipp.receivetransfer.TransfersInSectionCompleted.TransfersInSectionCompletedUserAnswersOps
+import play.api.libs.json.{Json, Reads}
+import pages.nonsipp.membersurrenderedbenefits._
+import models.{IdentityType, UserAnswers}
+import models.requests.DataRequest
+import pages.nonsipp.employercontributions.EmployerContributionsProgress.EmployerContributionsUserAnswersOps
+import models.softdelete.SoftDeletedMember
+import pages.nonsipp.employercontributions._
+import pages.nonsipp.membercontributions.TotalMemberContributionPage
+import pages.nonsipp.memberreceivedpcls.PensionCommencementLumpSumAmountPage
+import queries.{Gettable, Removable}
+import cats.syntax.option._
+import pages.nonsipp.membertransferout._
+
+import scala.util.{Failure, Try}
+
+trait SoftDelete { _: PSRController =>
+
+  protected def softDeleteMember(srn: Srn, index: Max300)(implicit req: DataRequest[_]): Try[UserAnswers] =
+    softDeleteMember(srn, index, req.userAnswers)
+
+  /**
+   * Only remove completed flags when the member is safe deleted
+   * return Option when something has gone wrong - todo: change to type that can store the error for logging
+   */
+  protected def softDeleteMember(srn: Srn, index: Max300, userAnswers: UserAnswers): Try[UserAnswers] = {
+    val ua = userAnswers
+    def get[A: Reads](f: (Srn, Max300) => Gettable[A]): Option[A] = ua.get(f(srn, index))
+
+    val memberDetails: Option[MemberPersonalDetails] = (
+      get(MemberDetailsPage),
+      get(NoNINOPage).some,
+      get(MemberDetailsNinoPage).map(_.value).some
+    ).mapN(
+      (nameDob, noNino, nino) => MemberPersonalDetails(nameDob.firstName, nameDob.lastName, nino, noNino, nameDob.dob)
+    )
+
+    val memberDetailsPages: List[Removable[_]] = List(
+      MemberDetailsPage(srn, index),
+      DoesMemberHaveNinoPage(srn, index),
+      NoNINOPage(srn, index),
+      MemberDetailsNinoPage(srn, index)
```

Comment:
There may be pages that are missing from here, when I look at the data after the soft delete, some are still there. E.g. memberStatus, reportAnotherTransferIn, totalMemberContribution, memberLumpSumReceived, pensionAmountReceived

Reply:
will add 👍

Reply:
added the extra pages to remove

Reply:
Looking good, all are now removed as far as I see, except "employerContributionsCompleted". Should that be removed too?